src/main/java/com/example/service/impl/UserServiceImpl.java

```diff
@@ -29,6 +32,9 @@ public UserDto register(UserRegistrationRequestDto userRequest) {
         user.setRoles(Set.of(roleRepository.findByName(Role.RoleName.USER).get()));
         user.setPassword(passwordEncoder.encode(userRequest.password()));
         User savedUser = userRepository.save(user);
+        ShoppingCart shoppingCart = new ShoppingCart();
```

Comment:
It's better to move this logic in ShoppingCartService and call this method here
`public void createShoppingCart(User user)`

Reply:
done